tests/countries/test_guinea.py

```diff
@@ -0,0 +1,217 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS.md file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.countries.guinea import Guinea, GN, GIN
+from tests.common import CommonCountryTests
+
+
+class TestGuinea(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        years = range(1959, 2050)
+        super().setUpClass(Guinea, years=years)
+        cls.no_estimated_holidays = Guinea(years=years, islamic_show_estimated=False)
+
+    def test_country_aliases(self):
+        self.assertAliases(Guinea, GN, GIN)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(Guinea(years=1958))
+
+    def test_new_years_day(self):
+        self.assertHolidayName(
+            "Fête du Nouvel an", (f"{year}-01-01" for year in range(1959, 2050))
+        )
+
+    def test_second_republic_day(self):
+        name = "Jour de la Deuxième République"
+        self.assertHolidayName(name, (f"{year}-04-03" for year in range(1959, 2022)))
+        self.assertNoHolidayName(name, range(2022, 2050))
```

Comment:
Test case change to be in line with @KJhellico 's Guinea's 2nd Republic year check above
```suggestion
    def test_second_republic_day(self):
        name = "Jour de la Deuxième République"
        self.assertHolidayName(name, (f"{year}-04-03" for year in range(1985, 2022)))
        self.assertNoHolidayName(name, range(1959, 1985), range(2022, 2050))
```